src/ophyd_async/core/_signal.py

```diff
@@ -113,8 +113,11 @@ def close(self):
 
     async def get_reading(self) -> Reading[SignalDatatypeT]:
         await self._valid.wait()
-        assert self._reading is not None, "Monitor not working"
-        return self._reading
+        if self._reading:
```

Comment:
Just to make it crystal clear that we are checking for None rather than an empty dict...
```suggestion
        if self._reading is not None:
```